cg/apps/sequencing_metrics_parser/parsers/bcl2fastq.py

```diff
@@ -16,93 +19,175 @@
+
 
+def update_lane_metrics_with_undetermined_tile_data(
+    lane_metrics: SampleLaneMetrics, undetermined_tile_metrics: Undetermined
+):
+    tile_total_reads: int = undetermined_tile_metrics.tile_total_reads
+    lane_metrics.total_reads += scale_paired_reads_to_total_reads(tile_total_reads)
 
-def aggregate_tile_metrics_per_sample_and_lane(
-    tile_metrics: List[Bcl2FastqSampleLaneTileMetrics],
-) -> List[Bcl2FastqSampleLaneMetrics]:
-    """Aggregate the tile metrics per sample and lane instead."""
+    tile_total_yield: int = undetermined_tile_metrics.tile_total_yield
+    lane_metrics.total_yield += tile_total_yield
+
+    tile_read_metrics: List[ReadMetric] = undetermined_tile_metrics.tile_read_metrics
+    lane_metrics.total_yield_q30 += sum_q30_yields(tile_read_metrics)
+    lane_metrics.total_quality_score += sum_quality_scores(tile_read_metrics)
+
+
+def combine_tiles_per_lane(tile_metrics: List[SampleLaneTileMetrics]) -> List[SampleLaneMetrics]:
+    """Aggregate the tile metrics to per lane instead."""
     metrics = {}
 
     for tile_metric in tile_metrics:
         for conversion_result in tile_metric.conversion_results:
-            for demux_result in conversion_result.demux_results:
-                sample_lane_key = (
+            for demux_result in conversion_result.tile_demux_results:
+                metric_key = (
                     conversion_result.lane_number,
                     demux_result.sample_id,
                 )
+                sample_id: str = remove_index_from_sample_id(demux_result.sample_id)
 
-                sample_id: str = discard_index_sequence(sample_id_with_index=demux_result.sample_id)
-
-                if sample_lane_key not in metrics:
-                    metrics[sample_lane_key] = Bcl2FastqSampleLaneMetrics(
+                if metric_key not in metrics:
+                    metrics[metric_key] = create_empty_lane_metric(
                         flow_cell_name=tile_metric.flow_cell_name,
-                        flow_cell_lane_number=conversion_result.lane_number,
+                        lane=conversion_result.lane_number,
                         sample_id=sample_id,
-                        sample_total_reads_in_lane=0,
-                        sample_total_yield_in_lane=0,
-                        sample_total_yield_q30_in_lane=0,
-                        sample_total_quality_score_in_lane=0,
                     )
-                # Double the total reads since they are reported in pairs
-                metrics[sample_lane_key].sample_total_reads_in_lane += demux_result.number_reads * 2
-                metrics[sample_lane_key].sample_total_yield_in_lane += demux_result.yield_
-                metrics[sample_lane_key].sample_total_yield_q30_in_lane += sum(
-                    [read_metric.yield_q30 for read_metric in demux_result.read_metrics]
+                update_lane_metrics_with_demux_data(
+                    lane_metrics=metrics[metric_key], demux_result=demux_result
```

Comment:
So it does not matter if we are updating or creating from scratch? Update_lane_metrics_with_demux_data will work either way?

Reply:
Yes, it will work either way since there is always an initialised metrics object to update.

Reply:
Renamed it to `update_lane_metrics_with_tile_reads`